src/ucp/core/ucp_context.h

```diff
@@ -13,7 +13,77 @@
 #include <uct/api/uct.h>
 #include <ucs/datastruct/queue_types.h>
 #include <ucs/type/component.h>
-
+#include <ucs/type/spinlock.h>
+#include "config.h"
+
+typedef struct ucp_mt_config {
+    int                           mt_required;
```

Comment:
can be consolidated into one field (enum() as follows:
{ NONE, SPINLOCK, MUTEX }